test/unit/fixes.py

```diff
@@ -2,12 +2,76 @@
 # Licensed under the MIT License.
 
 import sklearn
+from sklearn.utils.validation import check_array
 
 
 def get_sklearn_expected_1d_message():
     # Handle change of message for sklearn
     if sklearn.__version__ < "0.23.0":
         expected_message = "bad input shape"
     else:
-        expected_message = "y should be a 1d array"
+        expected_message = "Data must be 1-dimensional"
     return expected_message
+
+
+def validate_data(estimator, X, **kwargs):
+    """
+    Validate data for sklearn estimators. This function provides compatibility across
+    different sklearn versions by handling the validation method changes between versions.
+
+    Parameters
+    ----------
+    estimator : object
+        Sklearn estimator instance
+    X : array-like
+        Input data to validate
+    **kwargs : dict
+        Additional keyword arguments passed to the validation function
+
+    Returns
+    -------
+    array-like
+        The validated input data
+
+    Notes
+    -----
+    For sklearn versions >= 1.6, uses estimator._validate_data()
+    For sklearn versions < 1.6, uses sklearn.utils.validation.validate_data()
+    """
+    try:
+        from sklearn.utils.validation import validate_data
+
+        return validate_data(estimator, X, **kwargs)
+    except ImportError:
+        # sklearn version < 1.6
+        return estimator._validate_data(X, **kwargs)
+
+
+def _check_array(X, **kwargs):
```

Comment:
why call it `_check_array`. I'd call it `check_array` so that only the `import` statement in other places is changed and not the actual code.

Reply:
yeah this was def a questionable decision I agree, I wanted to leave obvious breadcrumbs for others as this is a very used function, but I can leave a comment at import level